kedro_neptune/__init__.py

```diff
@@ -80,99 +107,130 @@ def init(metadata: ProjectMetadata, api_token: str, project: str, base_namespace
 
     if not context.credentials_file.exists():
         with context.credentials_file.open("w") as credentials_file:
-            yaml.dump({
-                'neptune': {
-                    'NEPTUNE_API_TOKEN': api_token
-                }
-            },
-                credentials_file, default_flow_style=False)
+            credentials_file.writelines(INITIAL_NEPTUNE_CREDENTIALS.format(
+                api_token=api_token
+            ))
 
     context.config_file = context.project_path / settings.CONF_ROOT / config / "neptune.yml"
 
     if not context.config_file.exists():
         with context.config_file.open("w") as config_file:
-            yaml.dump({
-                'neptune': {
-                    'base_namespace': base_namespace,
-                    'project': project,
-                    'upload_source_files': ['**/*.py', f'{config}/base/*.yml']
-                }
-            }, config_file, default_flow_style=False)
+            config_file.writelines(INITIAL_NEPTUNE_CONFIG.format(
+                project=project,
+                base_namespace=base_namespace,
+                config=config
+            ))
 
+    context.catalog_file = context.project_path / settings.CONF_ROOT / config / "catalog_neptune.yml"
 
-class NeptuneMetadataDataSet(AbstractDataSet):
-    def __init__(self, run_id: str):
-        self._run_id: str = run_id
-        self._base_namespace: str = 'kedro'
-        self._run: Optional[neptune.Run] = None
+    if not context.catalog_file.exists():
+        with context.catalog_file.open("w") as catalog_file:
+            catalog_file.writelines(INITIAL_NEPTUNE_CATALOG)
 
-    def _save(self, data: Dict[str, Any]) -> None:
-        assert self._run is not None
 
-        self._run.assign(data)
+def get_neptune_config():
+    session: KedroSession = get_current_session()
+    context = session.load_context()
+    # pylint: disable=protected-access
+    credentials = context._get_config_credentials()
+    config = context.config_loader.get('neptune**')
+
+    api_token = credentials['neptune']['NEPTUNE_API_TOKEN']
+    project = config['neptune']['project']
+    base_namespace = config['neptune']['base_namespace'] or 'kedro'
```

Comment:
why this default?

Reply:
You mean `kedro` as default? It was included in the specification.

Reply:
point is you're already assigning `kedro` to `config['neptune']['base_namespace']` in `init()`, and if `config` is a normal `dict` (is it?), `or` won't help you with missing keys

so I'm not sure what scenario it handles